tests/torchtune/modules/peft/test_dora.py

```diff
@@ -221,3 +230,175 @@ def test_quantized_state_dict(self, dtype):
         assert torch.allclose(
             dora_linear.weight.quantized_data, dora_linear_reload.weight.quantized_data
         )
+
+    def test_dora_single_device_init(self, dora_linear):
+        dora_linear = dora_linear(
+            use_bias=False, dtype=torch.float32, should_init=False
+        )
+
+        # Randomly initialize LoRA A and B weights to some nonzero value
+        dora_linear.lora_a.weight = nn.Parameter(
+            torch.randn_like(dora_linear.lora_a.weight)
+        )
+        dora_linear.lora_b.weight = nn.Parameter(
+            torch.randn_like(dora_linear.lora_b.weight)
+        )
+
+        expected_magnitude = torch.linalg.norm(
+            dora_linear.weight
+            + dora_linear.scaling
+            * dora_linear.lora_b.weight
+            @ dora_linear.lora_a.weight,
+            dim=1,
+        )
+        assert not torch.allclose(dora_linear.magnitude, expected_magnitude)
+        dora_linear.initialize_dora_magnitude()
+        assert torch.allclose(dora_linear.magnitude, expected_magnitude)
+
+
+class TestDistributedDoRALinear(FSDPTest):
```

Comment:
This is very nice

Is it worth adding a test to make sure the `ValueError`occurs when we try to initialize dora magnitudes on meta device?

Reply:
Yeah I can add this